File: ts/src/algorithms/mixed/missing-number/missing.ts
export function missingNumber(nums: number[]): number {
  let missing = -1;
  let numsSet = new Set([...nums]);
  for (let i = 0; i <= nums.length; i++) {
    if (!numsSet.has(i)) {
      missing = i;
    }
  }
  return missing;
}

File: go/src/algorithms/mixed/missing-number/missing.go
package mixed

func missingNumber(nums []int) int {
	var store = make(map[int]bool)
	for _, n := range nums {
		store[n] = true
	}
	for i := 0; i <= len(nums); i++ {
		if _, ok := store[i]; !ok {
			return i
		}
	}
	return -1
}

// Using the formula for the sum of the first n numbers
// sum = n*(n+1)/2 - sum = missing number
func missingNumberV2(nums []int) int {
	var sum int
	for i := 0; i < len(nums); i++ {
		sum += nums[i]
	}
	return len(nums)*(len(nums)+1)/2 - sum
}
